### PR TITLE
Added more checks to RequestContext_LegacyActivityId_Simple

### PR DESCRIPTION
**Pull request by jason-bragg**
RequestContextTests_Silo.RequestContext_LegacyActivityId_Simple seems to be failing intermittently.  It's unclear if the issue is on the client or silo side.  This PR adds more checks to the test to help identify root cause, as the issue does not reproduce locally.

### CODE REVIEW

File: test/TestGrains/RequestContextTestGrain.cs [hunk 1]
```diff
@@ -40,6 +40,7 @@ public Task<Guid> E2EActivityId()
 
         public Task<Guid> E2ELegacyActivityId()
         {
+            if (!RequestContext.PropagateActivityId) throw new OrleansException("ActivityId Propegation is not enabled on silo.");
```
Comment (jdom) [inline test/TestGrains/RequestContextTestGrain.cs]: typo in `Propegation`

Comment (jdom) [reply test/TestGrains/RequestContextTestGrain.cs]: BTW, this should be InvalidOperationException or some other non-Orleans runtime exception, right?

**Comment by jason-bragg** [comment]
corrections LG2M

**Comment by jdom** [comment]
@dotnet-bot test functional
Reason: non related failure: https://ci.dot.net/job/dotnet_orleans/job/master/job/functional_prtest/377/

**Comment by jason-bragg** [comment]
@dotnet-bot test functional

**Pull request merged**